From 9a86952570b925e68dfd30a12642000353242745 Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Thu, 10 Oct 2024 17:16:08 +0100
Subject: [PATCH 1312/1350] Reapply "dtoverlays: Convert SenseHAT overlays to
 use MFD and upstream drivers"

This reverts commit 82a50e430ef1d6eb37d78e25aa572c1f6ea56160.
---
 .../boot/dts/overlays/rpi-sense-overlay.dts   | 28 +++++++++++++++++--
 .../dts/overlays/rpi-sense-v2-overlay.dts     | 28 +++++++++++++++++--
 2 files changed, 50 insertions(+), 6 deletions(-)

--- a/arch/arm/boot/dts/overlays/rpi-sense-overlay.dts
+++ b/arch/arm/boot/dts/overlays/rpi-sense-overlay.dts
@@ -12,11 +12,23 @@
 			#size-cells = <0>;
 			status = "okay";
 
-			rpi-sense@46 {
-				compatible = "rpi,rpi-sense";
+			sensehat@46 {
+				compatible = "raspberrypi,sensehat";
 				reg = <0x46>;
-				keys-int-gpios = <&gpio 23 1>;
+				interrupt-parent = <&gpio>;
 				status = "okay";
+
+				display {
+					compatible = "raspberrypi,rpi-sense-fb";
+					status = "okay";
+				};
+				joystick {
+					compatible = "raspberrypi,sensehat-joystick";
+					interrupts = <23 1>;
+					pinctrl-names = "default";
+					pinctrl-0 = <&sensehat_pins>;
+					status = "okay";
+				};
 			};
 
 			lsm9ds1-magn@1c {
@@ -44,4 +56,14 @@
 			};
 		};
 	};
+
+	fragment@1 {
+		target = <&gpio>;
+		__overlay__ {
+			sensehat_pins: sensehat_pins {
+				brcm,pins = <23>;
+				brcm,function = <0>;
+			};
+		};
+	};
 };
--- a/arch/arm/boot/dts/overlays/rpi-sense-v2-overlay.dts
+++ b/arch/arm/boot/dts/overlays/rpi-sense-v2-overlay.dts
@@ -12,11 +12,23 @@
 			#size-cells = <0>;
 			status = "okay";
 
-			rpi-sense@46 {
-				compatible = "rpi,rpi-sense";
+			sensehat@46 {
+				compatible = "raspberrypi,sensehat";
 				reg = <0x46>;
-				keys-int-gpios = <&gpio 23 1>;
+				interrupt-parent = <&gpio>;
 				status = "okay";
+
+				display {
+					compatible = "raspberrypi,rpi-sense-fb";
+					status = "okay";
+				};
+				joystick {
+					compatible = "raspberrypi,sensehat-joystick";
+					interrupts = <23 1>;
+					pinctrl-names = "default";
+					pinctrl-0 = <&sensehat_pins>;
+					status = "okay";
+				};
 			};
 
 			lsm9ds1-magn@1c {
@@ -44,4 +56,14 @@
 			};
 		};
 	};
+
+	fragment@1 {
+		target = <&gpio>;
+		__overlay__ {
+			sensehat_pins: sensehat_pins {
+				brcm,pins = <23>;
+				brcm,function = <0>;
+			};
+		};
+	};
 };
